llvm/unittests/Transforms/IPO/DynamicallyLoadedInliningAdvisor.cpp

```diff
@@ -0,0 +1,96 @@
+#include "llvm/AsmParser/Parser.h"
+#include "llvm/Config/config.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Testing/Support/Error.h"
+#include "gtest/gtest.h"
+
+extern llvm::cl::opt<std::string> DLInlineAdvisorPath;
+
+namespace llvm {
+
+void anchor() {}
+
+static std::string libPath(const std::string Name = "DAdvisor") {
+  const auto &Argvs = testing::internal::GetArgvs();
+  const char *Argv0 = Argvs.size() > 0 ? Argvs[0].c_str() : "PluginsTests";
+  void *Ptr = (void *)(intptr_t)anchor;
+  std::string Path = sys::fs::getMainExecutable(Argv0, Ptr);
+  llvm::SmallString<256> Buf{sys::path::parent_path(Path)};
+  sys::path::append(Buf, (Name + LLVM_PLUGIN_EXT).c_str());
+  return std::string(Buf.str());
+}
+
+TEST(DynamicInliningAdvisorTest, Foo) {
+  outs() << DLInlineAdvisorPath << '\n';
```

Comment:
check if this test should be run https://github.com/IBricchi/llvm-project/blob/ib/dlib-advisor-work/llvm/unittests/Passes/PluginsTest.cpp#L42